src/ssp.cpp

```diff
@@ -37,17 +41,28 @@ void ReadSSP(READ_SSP_ARGS)
     ssp->NPts = 1;
     
     for(int32_t iz=0; iz<MaxSSP; ++iz){
-        LIST_WARNLINE(ENVFile); ENVFile.Read(ssp->z[iz]); 
+        LIST_WARNLINE(ENVFile); ENVFile.Read(ssp->z[iz]);
         ENVFile.Read(RecycledHS.alphaR); ENVFile.Read(RecycledHS.betaR);
         ENVFile.Read(RecycledHS.rho);
         ENVFile.Read(RecycledHS.alphaI); ENVFile.Read(RecycledHS.betaI);
-        PRTFile << std::setprecision(2) << ssp->z[iz] << " " << RecycledHS.alphaR
-            << " " << RecycledHS.betaR << " " << RecycledHS.rho << " " 
-            << std::setprecision(4) << RecycledHS.alphaI << " " << RecycledHS.betaI << "\n";
+
+        if (resetParams) {
+            PRTFile << std::setprecision(2) << ssp->z[iz] << " " << RecycledHS.alphaR
+                << " " << RecycledHS.betaR << " " << RecycledHS.rho << " "
+                << std::setprecision(4) << RecycledHS.alphaI << " " << RecycledHS.betaI << "\n";
+            ssp->rho[iz] = RecycledHS.rho;
+            ssp->alphaR[iz] = RecycledHS.alphaR;
+            ssp->alphaI[iz] = RecycledHS.alphaI;
+        } else {
+            GlobalLog("Using existing sound speed profile: %g %g.\n",
+                ssp->alphaR[iz], ssp->alphaI[iz]);
+            RecycledHS.alphaR = ssp->alphaR[iz];
+            RecycledHS.alphaI = ssp->alphaI[iz];
+            // I'm not sure what to do with rho (density) and beta -- JS
+        }
```

Comment:
It seems like what you've done is, saved part of the SSP and then restored it when setup is called again without resetParams. I thought that what we were going to do was move all of the post-processing which is done in functions called by `setup` into functions called by `run` (or a separate external API function if necessary) so that you could edit the values in between. This is tricky for some things like converting km into meters--it would have to store somewhere whether the conversion had been done already, otherwise you could easily accidentally run the conversion multiple times.

Reply:
Yes, pulling out the iteratively updated parameters turned out to be 'tricky'. This has the advantage of working and being pretty simple, although that sounds like the definition of a hack:) I'll try pulling it out (after removing the logging) and see if the results at least still agree in the test cases.